packages/edit-site/src/classic-stylebook.js

```diff
@@ -0,0 +1,141 @@
+import SidebarNavigationScreen from './components/sidebar-navigation-screen/index.js';
+import {
+	StyleBookBody,
+	getExamplesForSinglePageUse,
+} from './components/style-book';
+import { getExamples } from './components/style-book/examples';
+
+const { RouterProvider } = unlock( routerPrivateApis );
+const { GlobalStylesProvider } = unlock( editorPrivateApis );
+
+function ClassicStylebookLayout() {
+	const [ resizeObserver, sizes ] = useResizeObserver();
+	const settings = useSelect(
+		( select ) => select( blockEditorStore ).getSettings(),
+		[]
+	);
+
+	const { colors, gradients } = useMultipleOriginColorsAndGradients();
+	// Exclude the default colors and gradients.
+	const themeColors = colors?.filter( ( color ) => color.slug === 'theme' );
+	const themeGradients = gradients?.filter(
+		( gradient ) => gradient.slug === 'theme'
+	);
+
+	const examples = getExamples( {
+		colors: themeColors,
+		gradients: themeGradients,
+		duotones: [], // Classic themes don't support duotone palettes.
+	} );
+
+	// Dedupe the examples as they include all categories with repeat sections.
+	const examplesForSinglePageUse = getExamplesForSinglePageUse( examples );
+
+	const route = {
+		name: 'stylebook',
+		areas: {
+			sidebar: (
+				<SidebarNavigationScreen
+					title={ __( 'Styles' ) }
+					description={ __( 'Overview of styled blocks.' ) }
+					isRoot
+				/>
+			),
+			preview: (
+				<StyleBookBody
+					enableResizing={ false }
+					showCloseButton={ false }
+					showTabs={ false }
+					examples={ examplesForSinglePageUse }
+					settings={ settings }
+					sizes={ sizes }
+				/>
+			),
+		},
+	};
+	return (
+		<>
+			{ resizeObserver }
+			<Layout route={ route } />
+		</>
+	);
+}
+
+/**
+ * Initializes the classic stylebook.
+ * @param {string} id       ID of the root element to render the screen in.
+ * @param {Object} settings Editor settings.
+ */
+export function initializeClassicStylebook( id, settings ) {
+	if ( ! globalThis.IS_GUTENBERG_PLUGIN ) {
+		return;
+	}
+	const target = document.getElementById( id );
+	const root = createRoot( target );
+
+	dispatch( blocksStore ).reapplyBlockTypeFilters();
+	const coreBlocks = __experimentalGetCoreBlocks().filter(
+		( { name } ) => name !== 'core/freeform'
+	);
+	registerCoreBlocks( coreBlocks );
+	dispatch( blocksStore ).setFreeformFallbackBlockName( 'core/html' );
+	registerLegacyWidgetBlock( { inserter: false } );
+	registerWidgetGroupBlock( { inserter: false } );
+	if ( globalThis.IS_GUTENBERG_PLUGIN ) {
+		__experimentalRegisterExperimentalCoreBlocks( {
+			enableFSEBlocks: true,
+		} );
+	}
+
+	dispatch( editSiteStore ).updateSettings( settings );
+
+	root.render(
+		<StrictMode>
+			<GlobalStylesProvider>
+				<UnsavedChangesWarning />
+				<RouterProvider>
+					<BlockEditorProvider settings={ settings }>
+						<ClassicStylebookLayout />
+					</BlockEditorProvider>
+				</RouterProvider>
+			</GlobalStylesProvider>
+		</StrictMode>
+	);
+
+	return root;
+}
```

Comment:
> Why is this creating a separate page rather than reusing the site editor

The assumption here is that it's preferable to use the existing site-editor.php page? 

E.g., directly linking from the admin bar to `/wp-admin/site-editor.php?postType=wp_global_styles`

If it cuts down on future maintenance and reduces boilerplate code, it sounds like a good idea.

I'm guessing we'd need to tweak the feature access conditions. E.g., rolling in a [wp_theme_has_theme_json](https://github.com/WordPress/wordpress-develop/blob/8ab940fed762f16f9f92f2c6c4f4cb560ee68366/src/wp-includes/global-styles-and-settings.php#L399-L399) check  (?) 

```php
$is_global_styles_path   = isset( $_GET['path'] ) && 'wp_global_styles' === sanitize_key( $_GET['path'] );

if ( ! wp_is_block_theme() ) {
	if ( ( ! wp_theme_has_theme_json() && $is_global_styles_path ) || ( ! current_theme_supports( 'block-template-parts' ) && $is_template_part_editor ) ) {
		wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
	} elseif ( ! $is_global_styles_path && ! $is_patterns_editor && ! $is_template_part_editor ) {
		wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
	}
}
```

Loads kinda, but the site editor nav and the layout would probably need some condition boundaries too. 

While sniffing about, I noticed there are a [couple of related editor settings](https://github.com/WordPress/wordpress-develop/blob/44e6c75ff94b6f05ce8dd404237b1a2badf8e8e9/src/wp-admin/site-editor.php#L66-L67):

- `supportsLayout`, which is the output of `wp_theme_has_theme_json()` anyway, and 
- `supportsTemplatePartsMode`, which, as far as I can tell, is not used anywhere in today's Gutenberg or Core.


> Also, long term it might be good to make that condition overridable with a filter or something

That makes sense to me - it would make developing from Gutenberg more flexible.

Reply:
yes, my main thinking is that I don't want us to create two separate UIs/pages. It's ok to have two different entry points if needed, I just want to share the screens themselves.

Reply:
> I'm guessing we'd need to tweak the feature access conditions. E.g., rolling in a [wp_theme_has_theme_json](https://github.com/WordPress/wordpress-develop/blob/8ab940fed762f16f9f92f2c6c4f4cb560ee68366/src/wp-includes/global-styles-and-settings.php#L399-L399) check (?)

We need more than that, because the stylebook has to be available to classic themes without theme.json too. The end goal is:
* if there's no theme.json, show a static stylebook
* if there's a theme.json with some styles defined, also show global styles controls to edit those styles.

I think it does make sense to not create a separate UI, but I can't work out how to do that given the current core condition prevents the site editor from loading when classic themes are active. If anyone has concrete suggestions for a workaround that doesn't involve a separate UI, they are very welcome!

Otherwise, I'd suggest we go ahead with a _temporary_ separate UI, and update it to use the site editor once core is updated.

Reply:
I think you're already doing that to a point. Basically `ClassicStylebookLayout` should ideally not exist and we should just render the SiteEditor layout instead as if the regular "init" function of the site editor

Reply:
Ok I updated this draft to use the SiteEditor layout with a dedicated `classicStylebookRoute` to render the correct stuff in it. I'm thinking that for classic themes with theme.json we could maybe have a different route that includes global styles in the content area.

I did have to create a new `StyleBookPreview` wrapper to circumvent `StyleBook`'s editor dependency, because Editor component refuses to render without a post id 😅 
The other useful thing `StyleBookPreview` does is fetch the correct palettes for a classic theme (where they exist - not all themes declare them)

Not sure if there's a better way to do this.

Further feedback welcome!

Reply:
> Can't we use the same StyleBookPreview component for both?

Good question, I'm not sure why block theme style book uses the editor. It may just be an artifact of its original location, and reusing the `StyleBook` component in the preview means the editor is part of the package. I can try switching it out and see how it goes!

It might mean that the logic for fetching the color palettes has to look up what type of theme is active.

Reply:
> Is this route necessary because we want to avoid the "content" area entirely?

Yeah, I can't see any way around this. I tried adding a check to conditionally render a content area in the route, but at that point the block editor settings are still the default ones, so there's no way of knowing whether a theme.json exists or not (we don't want to render the content area/global styles UI if there's no theme.json, regardless of the theme being classic or block).

I'm also not sure how to tailor the URL to match the global styles one: `path=/wp_global_styles` because the URL I'm getting from `add_theme_page` (or `add_submenu_page`) is always structured like `page=gutenberg-classic-stylebook`. There may be some workaround for this I don't know about 😅

Ideas welcome!

Reply:
> I'm not sure why block theme style book uses the editor

Update on this: I think I see why this was done. `StyleBook` is rendering inside `GlobalStylesUIWrapper`, which is in the content area. So it's using `EditorCanvasContainer` to render its contents in the `EditorContentSlotFill`, which puts it in the preview area.

I'm not quite sure how to untangle this 😅 or even if it's worth untangling before we try moving the global styles navigation into the black sidebar, because that'll shift the areas being occupied a little.

I think for now let's focus on displaying a static stylebook for classic themes without theme.json. For next steps, there's a few things we can do:

* Add the global styles UI for classic themes with theme.json (as it stands this will likely involve a new route, otherwise we can do it after we improve the block theme stylebook)
* Improve global styles UI to use black sidebar for navigation and the content area for any submenus
* Consolidate block theme stylebook to not rely on the Editor component
* It would also be nice if the block theme stylebook view had its own URL, because right now it's impossible to link to it directly.

Reply:
Thanks for investigating and documenting all this, folks.

> StyleBook is rendering inside GlobalStylesUIWrapper, which is in the content area. So it's using EditorCanvasContainer to render its contents in the EditorContentSlotFill, which puts it in the preview area.

From memory — I'd have to channel Rob — the style book [was added as a slot inside the editor](https://github.com/WordPress/gutenberg/pull/45960/files#diff-356ea6205e41f6b1c7ac6fbc3b4c4f08c07cef01a30ff6e3c015762cd0568d4eR234) as a matter of convenience. In the editor context, all the block styles are loaded, block editor settings and global styles provider etc.

> Consolidate block theme stylebook to not rely on the Editor component

I think that's been a goal for a while, at least to enable swapping it out for the editor, instead jamming it in via the [<EditorContentSlotFill.Slot>](https://github.com/WordPress/gutenberg/blob/0db7ff64bb9263ebffa23efb0e3c5aae5668ebbe/packages/editor/src/components/editor-interface/index.js#L154-L154). It's at least what I was trying to do over in https://github.com/WordPress/gutenberg/pull/66848/files

I'd also hoped that an eventual side-effect of removing this slot would be to remove the `editorCanvasView` store that determines the slot's contents.

To be honest, I'm not sure how to neatly untangle it either quite yet.

> I think for now let's focus on displaying a static stylebook for classic themes without theme.json

 👍🏻 Sounds good! We don't have to solve everything right now. 

I suppose it'd be neat not to have the change the URL later, say, if we do consolidate and have everything under `/wp-admin/site-editor.php?path=/wp_global_styles`, but one thing at a time. 😄 

> It would also be nice if the block theme stylebook view had its own URL, because right now it's impossible to link to it directly.

Something like `/wp-admin/site-editor.php?path=/wp_global_styles/style-book`?

Reply:
> The obstacle to this is that nothing under `site-editor.php` can be accessed from a classic theme until we change core.

Could filtering `wp_die()` be the solution to this?

I ran into a similar issue when exposing the Patterns page for a classic theme.

https://github.com/WordPress/gutenberg/pull/60804/files#diff-9e66dbabd964e0d41f3dec92b7177ec29e9e5f5903dc68a206e85622e1f6058eR74-R92

Reply:
Oh interesting approach, I hadn't thought of that! I'll give it a try.

Reply:
Done! This also allowed me to get rid of the separate init function and styles, so now everything is being reused from the site editor itself 🎉 

Thanks for the idea @t-hamano!